.github/workflows/ci.yml

```diff
@@ -0,0 +1,68 @@
+name: ci
+
+# this build will only trigger automatically when `master` pushes
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  ci:
+    name: test-${{ matrix.os }}-python${{ matrix.python-version }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+
+        # this works like a loop; think `pytest.mark.parametrize`
+        python-version: [2.7, 3.5, 3.6, 3.7, 3.8]
+        os: [ubuntu-latest, windows-latest, macos-latest]
+
+    steps:
+    - name: checkout build
+      id: checkout-build
+      uses: actions/checkout@v2
+
+    # with this we do not need to worry about the various ways in which
+    # each os installs python
+    - name: setup python
+      id: setup-python
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: upgrade pip
+      id: upgrade-pip
+      run: |
+        python -m pip install --upgrade pip
+
+    # gets the path to wherever each os stores pip's cache
+    - name: get pip cache dir
+      id: get-pip-cache-dir
+      run: |
+        echo "::set-output name=dir::$(pip cache dir)"
+
+    - name: cache pip
+      uses: actions/cache@v2
+      id: cache-pip
+      with:
+
+        # set above with `get pip cache dir`
+        path: ${{ steps.get-pip-cache-dir.outputs.dir }}
+
+        # determines whether the hash of setup.py has changed
+        # CACHE_VERSION: slight hack, see https://stackoverflow.com/a/64819132
+        key: ${{ runner.os }}-pip-${{ secrets.CACHE_VERSION }}-${{ hashFiles('**/setup.py') }}
+
+    - name: install packages
+      id: install-packages
+      run: |
+        python -m pip install .[dev] codecov
+
+    - name: test
+      id: test
+
+      # `tox-gh-actions` works out a mapping for `matrix.python-version`
+      # e.g. `python-version: [3.6]` -> `tox -e py36`
+      # python -m codecov only to rn if exit status is 0
+      run: |
+        python -m tox -- --cov-report=xml && python -m codecov
```

Comment:
Each job is executed independently across OS / Python versions right? If so, what's the benefit of using `tox` to run the jobs?

Reply:
You're right, I think we can just call pytest. I will look into the python runner, I will double check how it interacts with the `python - version` array. That would benefit us in not having to make any changes to the Pipfile, Pipfile.lock, and tox.ini files.

Reply:
Sweet, yeah I could go either way on it. On the one hand, just calling `tox` would simplify the virtual environment setup/installation; it would also exercise the `tox.ini` config to keep it tested. But on the other, the CI config is easier to read and maintain without the `tox` calls

Reply:
Yeah I think tbh the calls to tox may obfuscate what is actually happening in actions.